src/Microsoft.Azure.WebJobs.Host/Indexers/MethodInfoExtensions.cs

```diff
@@ -26,7 +28,15 @@ public static string GetShortName(this MethodInfo methodInfo)
                 throw new ArgumentNullException("methodInfo");
             }
 
-            return String.Format(CultureInfo.InvariantCulture, "{0}.{1}", methodInfo.DeclaringType.Name, methodInfo.Name);
+            return TryGetCustomName(methodInfo, out string customName)
+                ? customName
+                : String.Format(CultureInfo.InvariantCulture, "{0}.{1}", methodInfo.DeclaringType.Name, methodInfo.Name);
+        }
+
+        private static bool TryGetCustomName(MethodInfo methodInfo, out string customName)
```

Comment:
I only saw FunctionDescriptor being created in one place with a functionName, and that calls GetFullName and GetShortName. I might be wrong though.